### PR TITLE
Refactor/Improve specs readability

### PR DESCRIPTION
**Pull request by cassianoblonski**
### What changed, and why?
* Enforce that subject is the first definition in the test.
* Fix incorrect uses of context to specify methods.
* Change some descriptions
* Remove should syntax in favor of expect
* Use do end for multiline blocks `(example: let(:params) {... multilines ...} to  let(:params) do ... multilines ... end)`

#### Use the `disable_monkey_patching!` configuration option to disable all monkey patching done by RSpec:
* stops exposing DSL globally
* disables should and should_not syntax for rspec-expectations
* disables stub, should_receive, and should_not_receive syntax for rspec-mocks

### How is this tested? 💖💪
Running the full test suite.

### CODE REVIEW

**Comment by cassianoblonski** [comment]
@jmkoni @compwron 
Can you guys take a look here and review these before some conflicts show up?

File: spec/models/case_contact_spec.rb [hunk 1]
```diff
@@ -2,8 +2,8 @@
 
 RSpec.describe CaseContact, type: :model do
   context "validations" do
-    it { should validate_numericality_of(:miles_driven).is_less_than 10000 }
-    it { should validate_numericality_of(:miles_driven).is_greater_than_or_equal_to 0 }
+    it { is_expected.to validate_numericality_of(:miles_driven).is_less_than 10_000 }
+    it { is_expected.to validate_numericality_of(:miles_driven).is_greater_than_or_equal_to 0 }
```
Comment (jmkoni) [inline spec/models/case_contact_spec.rb:6]: Why this change? Not saying it's wrong, just curious.

File: spec/models/case_contact_spec.rb [hunk 2]
```diff
@@ -82,20 +82,20 @@
     expect(case_contact.errors[:base]).to eq(["cannot edit case contacts created before the current quarter"])
   end
 
-  context "#update_cleaning_contact_types" do
+  describe "#update_cleaning_contact_types" do
```
Comment (jmkoni) [inline spec/models/case_contact_spec.rb:85]: Same here... did you make this change because there's not a describe within context?

File: spec/models/user_spec.rb [hunk 3]
```diff
@@ -51,7 +52,7 @@
   end
 
   describe "supervisors" do
-    context "#volunteers_serving_transistion_aged_youth" do
+    describe "#volunteers_serving_transistion_aged_youth" do
```
Comment (jmkoni) [inline spec/models/user_spec.rb:55]: Yeah, can you clarify a bit more on why you changed some of these? What was your standard for `describe` over `context` and vice versa?

File: spec/models/volunteer_spec.rb [hunk 4]
```diff
@@ -75,11 +75,12 @@
   describe "#made_contact_with_all_cases_in_days?" do
     let(:volunteer) { create(:volunteer) }
     let(:casa_case) { create(:casa_case) }
-    let(:create_case_contact) {
-      ->(occurred_at, contact_made) {
+    let(:create_case_contact) do
+      lambda { |occurred_at, contact_made|
```
Comment (jmkoni) [inline spec/models/volunteer_spec.rb:79]: Not sure this change was necessary...

File: spec/requests/all_casa_admins_spec.rb [hunk 5]
```diff
@@ -45,14 +45,14 @@
     before { sign_in admin }
 
     context "with valid parameters" do
-      let(:params) {
+      let(:params) do
```
Comment (jmkoni) [inline spec/requests/all_casa_admins_spec.rb:48]: Don't think this is necessary either.

File: spec/requests/all_casa_admins_spec.rb [hunk 6]
```diff
@@ -62,14 +62,14 @@
     end
 
     context "with invalid parameters" do
-      let(:params) {
+      let(:params) do
```
Comment (jmkoni) [inline spec/requests/all_casa_admins_spec.rb:65]: Same here.

File: spec/requests/case_contact_reports_spec.rb [hunk 7]
```diff
@@ -21,12 +22,12 @@
 
     shared_examples "can view reports" do
       context "with start_date and end_date" do
-        let(:case_contact_report_params) {
+        let(:case_contact_report_params) do
```
Comment (jmkoni) [inline spec/requests/case_contact_reports_spec.rb:25]: Same here, unnecessary change.

File: spec/requests/volunteers_spec.rb [hunk 8]
```diff
@@ -36,16 +36,18 @@
     before do
       sign_in admin
     end
+
     context "with valid params" do
-      let(:params) {
+      let(:params) do
```
Comment (jmkoni) [inline spec/requests/volunteers_spec.rb:41]: Same

File: spec/requests/volunteers_spec.rb [hunk 9]
```diff
@@ -61,16 +63,18 @@
         }.to change { ActionMailer::Base.deliveries.count }.by(1)
       end
     end
+
     context "with invalid parameters" do
-      let(:params) {
+      let(:params) do
```
Comment (jmkoni) [inline spec/requests/volunteers_spec.rb:68]: Same

**Review by jmkoni — COMMENTED** [review]
I left a few comments, but I'm mostly unsure why this change is necessary...

File: spec/models/case_contact_spec.rb (hunk 1 above)
Comment (compwron) [reply spec/models/case_contact_spec.rb:6]: this is the new rspec style 
I have used https://www.betterspecs.org/#expect in the past and I believe it is one of the influences for this change.

File: spec/models/case_contact_spec.rb (hunk 2 above)
Comment (compwron) [reply spec/models/case_contact_spec.rb:85]: context and describe do the same thing, it's just a matter of which one "sounds" better
In general, describe is for methods, and context is for data setup- but it is not a hard and fast rule https://www.betterspecs.org/#describe

File: spec/models/volunteer_spec.rb (hunk 4 above)
Comment (compwron) [reply spec/models/volunteer_spec.rb:79]: I am happy with lambda over `->` as it is more google-able. it's a little odd to me that `standardrb` doesn't enforce it. As long as they are fully equivalent for test setup (i.e. the tests don't fail or falsely-pass) and the linter likes it, I am happy with the more searchable term.

File: spec/models/case_contact_spec.rb (hunk 2 above)
Comment (jmkoni) [reply spec/models/case_contact_spec.rb:85]: Yeah, that was my understanding. That's why I'm unsure as to why these were changed in this way.

File: spec/requests/all_casa_admins_spec.rb (hunk 5 above)
Comment (compwron) [reply spec/requests/all_casa_admins_spec.rb:48]: I usually see `do` used over `{` on multi-line blocks and I thought that standardrb enforced it. Is our linter skipping our specs? And if so, is that intentional? They're not in the ignore file

Comment (jmkoni) [reply spec/requests/all_casa_admins_spec.rb:48]: Not sure about that (re: standard). I'm not sure how much it digs into rspec yet. I don't really care, I'm just not in favor of making a change like this just because. But 🤷‍♀️ this is why I didn't mark them as blocking.

Comment (jmkoni) [reply spec/requests/all_casa_admins_spec.rb:48]: `let` and other rspec-isms are ignored by Standard's semantic blocks https://github.com/testdouble/standard/blob/master/config/base.yml#L745

File: spec/models/case_contact_spec.rb (hunk 2 above)
Comment (compwron) [reply spec/models/case_contact_spec.rb:85]: This usage of "describe" for a method is more correct than "context" for a method according to https://www.betterspecs.org/#describe above which I assume is the explanation :)

**Review by compwron — APPROVED** [review]
The vast majority of these changes look unequivocally positive! I am happy to review a follow-up PR to change any lingering issues. :)

**Comment by compwron** [comment]
You are invited to the CASA asynchronous retrospective-and-futurespective! Please take a look at https://bit.ly/casaretro (no login required) and add at least one note! Even if it's just "I was here and you are cool" or "omg y'all why do you keep messaging me" we want to hear it. <3
Next week Wednesday 18 November @ 6pm Pacific time we will Zoom about the board and talk about themes and takeaways. (Zoom link in our calendar at http://bit.ly/casacal or slack #casa channel https://rubyforgood.herokuapp.com/ )
Please ESPECIALLY add lots of notes if you won't be able to make it on Wednesday to talk!
* Note: the board will be monitored for spam; message me or any team lead if you see a problem
https://bit.ly/casaretro
